### PR TITLE
feat: :sparkles: setting to collapse advanced by default

### PR DESCRIPTION
**Pull request by joshistoast**
Adds a setting to keep advanced options open/closed by default within UI accordions

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Community Node Submission

## Have you discussed this change with the InvokeAI team?
- [ ] Yes
- [x] No, because: I thought this would be neat

      
## Have you updated all relevant documentation?
- [ ] Yes
- [x] No

## Description
Adds a neat setting for whether the "advanced options" accordions within parameter panel accordions should be expanded by default. This would be beneficial to InvokeAI power users who dislike opening more panels and such to find the settings they're looking for.

## QA Instructions, Screenshots, Recordings
image

## Added/updated tests?

- [ ] Yes
- [x] No : I'm not in love with setting up tests nor am I familiar with the existing testing methods.

### CODE REVIEW

**Comment by joshistoast** [comment]
There may be a better way to tell these parameter panels the default advanced expanded state outside of props...

File: invokeai/frontend/web/.eslintrc.js [hunk 1]
```diff
@@ -122,6 +122,12 @@ module.exports = {
         'i18next/no-literal-string': 'off',
       },
     },
+    {
+      files: ['**/*.ts', '**/*.tsx'],
+      rules: {
+        'react/prop-types': 'off',
+      },
+    },
```
Comment (psychedelicious) [inline invokeai/frontend/web/.eslintrc.js:130]: Uhh this should not be necessary

File: invokeai/frontend/web/src/features/settingsAccordions/components/GenerationSettingsAccordion/GenerationSettingsAccordion.tsx [hunk 2]
```diff
@@ -46,60 +46,65 @@ const badgesSelector = createMemoizedSelector(
...
-      defaultIsOpen: false,
-    });
-  const { isOpen: isOpenAccordion, onToggle: onToggleAccordion } =
-    useStandaloneAccordionToggle({
-      id: 'generation-settings',
-      defaultIsOpen: true,
-    });
+interface GenerationSettingsAccordionProps {
+  collapseAdvanced?: boolean;
+}
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/settingsAccordions/components/GenerationSettingsAccordion/GenerationSettingsAccordion.tsx:51]: If this is a global flag, to be applied to all of the expanders, it probably belongs in `useExpanderToggle`.

**Review by psychedelicious — CHANGES_REQUESTED** [review]
I don't think this is necessary.

I just implemented persistence for the accordions and expanders the other day. I don't think this new setting actually does anything:
- If you are starting the app without any persisted settings, the expanders start collapsed. This is the same behaviour as implemented in the PR.
- If you have changed the state of the expanders, that is remembered on next load, so the new setting is ignored.

So, there's no scenario in which the changes in this PR should affect the behaviour.

Is the persistence not working / am I missing something?

File: invokeai/frontend/web/.eslintrc.js (hunk 1 above)
Comment (joshistoast) [reply invokeai/frontend/web/.eslintrc.js:130]: Without it there's a prop validation error so 🤷‍♂️

**Comment by joshistoast** [comment]
> So, there's no scenario in which the changes in this PR should affect the behaviour.

Well it works, by default they all remain collapsed and toggling the setting opens them all unless closed.

File: invokeai/frontend/web/.eslintrc.js (hunk 1 above)
Comment (psychedelicious) [reply invokeai/frontend/web/.eslintrc.js:130]: It failed bc this is not correct:

```ts
memo<ImageSettingsAccordionProps>(({ collapseAdvanced = false }) => { ... })
```

memo's type argument is a component, not the props of a component.

It should be:
```ts
memo(({ collapseAdvanced = false }: ImageSettingsAccordionProps) => { ... })
```

**Comment by psychedelicious** [comment]
> Well it works, by default they all remain collapsed and toggling the setting opens them all unless closed.

It only works if you haven't manually opened the expanders yet. 

On first load, the expanders/accordions have a default `isOpen` state. Once the user opens or closes an expander/accordion for the first time, its state is persisted and the default state is always ignored.

This new setting changes only the default state. Once the user toggles an expander, this setting is ignored.

Closing in favour of the existing behaviour - if you want the expanders open, just open them once and they will remember their state.

**Pull request closed without merging**